packages/flutter/lib/src/services/text_input.dart

```diff
@@ -413,6 +417,48 @@ class TextInputConfiguration {
   /// Defaults to true.
   final bool autocorrect;
 
+  /// {@template flutter.services.textInput.enableSmartDashes}
+  /// Whether to allow the platform to automatically format dashes.
+  ///
+  /// This flag only affects iOS. It sets UITextSmartDashesType in the engine.
+  /// When true, it passes UITextSmartDashesTypeYes, and when false, it passes
+  /// UITextSmartDashesTypeNo.
+  ///
+  /// As an example of what this does, two consecutive hyphen characters will be
+  /// automatically replaced with one en dash, and three consecutive hyphens
+  /// will become one em dash.
+  ///
+  /// Defaults to true, unless obscureText is true, when it defaults to false.
+  /// This is to avoid the problem where password fields receive autoformatted
+  /// characters.
+  ///
+  /// See also:
+  ///  * [enableSmartQuotes]
+  ///  * <https://developer.apple.com/documentation/uikit/uitextinputtraits>
+  /// {@endtemplate}
+  final bool enableSmartDashes;
+
+  /// {@template flutter.services.textInput.enableSmartQuotes}
+  /// Whether to allow the platform to automatically format quotes.
+  ///
+  /// This flag only affects iOS. It sets UITextSmartQuotesType in the engine.
+  /// When true, it passes UITextSmartQuotesTypeYes, and when false, it passes
+  /// UITextSmartQuotesTypeNo.
```

Comment:
nit:
```dart 
 /// This flag only affects iOS. It sets `UITextSmartQuotesType` in the engine.
 /// When true, it passes `UITextSmartQuotesTypeYes`, and when false, it passes
 /// `UITextSmartQuotesTypeNo`.
```
I wonder if dart docs understands hyper links. Would be nice if we can link these uikit references to their api docs.

Reply:
Also doesn't it pass `UITextSmartQuotesTypeDefault` when true, according to the engine PR?

Reply:
Linked and updated to "default" (however, see my question above about "yes" vs. "default").